### ISSUE TITLE
VIDEOJS: ERROR: videojs-contrib-ads has not seen a loadstart event 5 seconds after being initialized, but a source is present.

### ISSUE THREAD

**Issue by Slind14** [videojs/videojs-contrib-ads#394]
I'm encountering the error `VIDEOJS: ERROR: videojs-contrib-ads has not seen a loadstart event 5 seconds after being initialized, but a source is present.` on a frequent basis, no matter the browser. I even put both files (video.js and videojs-contrip-ads) in a single file so that latency doesn't mess anything up. Unfortunately this error even happens in this case. Is there anything else that needs doing besides making sure they load in the right order?

**Comment by Slind14** [comment]
Yeah, I read that, hence I put them in a single file to make sure they are executed in the particular order without anything in between.
https://transfer.sh/%28/qXAbH/videojsads.zip%29.zip

**Comment by incompl** [comment]
Looking at `videojsadplayer.js`, the entire file is wrapped in a `ready`, including initialization of the ad plugin. You'll want that ad plugin to be initialized outside of the ready handler, since that is deferring it. Try something like this:

```js
const player = videojs("content_video"); // initializes player
player.ima(options, func); // immediately initialize ad plugin
player.ready(function() { // do things after player is ready
```

**Comment by Slind14** [comment]
that seems to have resolved it, thank you :)

**Comment by incompl** [comment]
Glad to help!

**Comment by damianvila** [comment]
This helped me a lot too! I owe you a cup of coffee @incompl :smile:
